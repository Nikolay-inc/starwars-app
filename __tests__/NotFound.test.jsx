import React from 'react';
import { render } from '@testing-library/react';
import NotFound from '../src/app/not-found';

describe('NotFound Component', () => {
  test('renders page not found message', () => {
    const { getByText } = render(<NotFound />);
    const headingElement = getByText(/Page not found/i);
    expect(headingElement).toBeInTheDocument();
  });

  test('renders resource not found message', () => {
    const { getByText } = render(<NotFound />);
    const paragraphElement = getByText(/Could not find requsted resourse/i);
    expect(paragraphElement).toBeInTheDocument();
  });

  test('renders not found wrapper', () => {
    const { getByTestId } = render(<NotFound />);
    const wrapperElement = getByTestId('not-found-wrapper');
    expect(wrapperElement).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { container } = render(
      <NotFound />
    );
    expect(container).toMatchSnapshot();
  });
});
