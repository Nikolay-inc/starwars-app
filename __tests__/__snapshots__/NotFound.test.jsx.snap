// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Component matches snapshot 1`] = `
<div>
  <div
    class="notFoundWrapper"
    data-testid="not-found-wrapper"
  >
    <h1>
      Page not found
    </h1>
    <p>
      Could not find requsted resourse
    </p>
  </div>
</div>
`;
